### PR TITLE
add to config

### PR DESCRIPTION
**Pull request by kordjamshidi**

### CODE REVIEW

**Comment by kordjamshidi** [comment]
please merge if this is ok

File: saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/mSpRLConfigurator.scala [hunk 1]
```diff
@@ -13,4 +15,5 @@ object mSpRLConfigurator {
   val expName = "BM" //{"BM", "BM+C", "BM+C+E", "BM+C+E+I"}
   val isTrain = true
   val iterations = 50
+  val useConstraints = false
```
Comment (Rahgooy) [inline saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/mSpRLConfigurator.scala:18]: what if the user chooses the "BM+C" but forgets about `useConstraints`?

Comment (Rahgooy) [reply saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/mSpRLConfigurator.scala:18]: how about `val useConstraints = expName.contains("+C")` ?

Comment (Rahgooy) [reply saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/mSpRLConfigurator.scala:18]: In the SRL app, you have gone the other way, defined the parameters in the configurator and the expName in the app. do you want to change this to follow that approach?

Comment (kordjamshidi) [reply saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/mSpRLConfigurator.scala:18]: yes, that was what I tried to suggest on slack, in my draft.
we can decide about the expname also in the config if you like, as follows:
```
 val expName: String = {
    if (!USE_CONSTRAINTS && !USE_EMBEDDINGS && !USE_IMAGE_EMBEDDINGS) "BM"
    else
    if (USE_CONSTRAINTS && !USE_EMBEDDINGS && !USE_IMAGE_EMBEDDINGS) "BM+C"
    else
    if (USE_CONSTRAINTS && USE_EMBEDDINGS && !USE_IMAGE_EMBEDDINGS) "BM+C+E"
    else
    if (!USE_CONSTRAINTS && !USE_EMBEDDINGS && !USE_IMAGE_EMBEDDINGS) "BM+C+E+I"
    else ""
    }
```

Comment (Rahgooy) [reply saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/mSpRLConfigurator.scala:18]: OK, I'll merge this. And change the app

Comment (kordjamshidi) [reply saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/mSpRLConfigurator.scala:18]: we have this ``` MultiModalSpRLDataModel.useVectorAverages = false``` in the App which shouls be integrated here as well.

**Pull request merged**